### ISSUE TITLE
Replace fragile Firefox workaround for <summary> arrow icon

### ISSUE THREAD

**Issue by robinwhittleton** [alphagov/govuk_elements#553]
The styling for `<summary>` on GOV.UK currently expects the focus ring to shrinkwrap the text. This works fine in Chrome, but fails in Firefox. I worked around this with some [horrendous code](https://github.com/alphagov/govuk_elements/blob/dea6e7bbaf83ecc670bc2cd31539096fab542484/assets/sass/elements/_shame.scss) but:

1. that just resets Firefox back to not shrinkwrapping, and
2. it turned out that the Firefox implementation is based on [the WHATWG suggested initial rendering](https://html.spec.whatwg.org/multipage/rendering.html#the-details-and-summary-elements) for summary.

Having read through I started worrying that other browsers might implement the WHATWG initial styles in the future, but a different way of fixing the problem did occur to me. While we can’t use `display: list-item` directly, we can include the list icons directly using counters. The following example code seems to work in Firefox and Chrome at least, although it’s untested beyond that and would need a proper accessibility run through etc:

```css
summary {
  display: inline-block;
  counter-reset: summary-closed;
}
summary::before {
  content: counter(summary-closed, disclosure-closed) " ";
}
details[open] > summary {
  counter-reset: summary-open;
}
details[open] > summary::before {
  content: counter(summary-open, disclosure-open) " ";
}
```

`disclosure-closed` and `disclosure-open` are the default markers used for `<summary>` elements. The `" "` just inserts an empty space inbetween the marker and the text.

Does this sound like a reasonable approach?

**Comment by 36degrees** [comment]
This was done in https://github.com/alphagov/govuk_elements/pull/611.
